src/System.Net.Http/src/netcore50/System/Net/HttpHandlerToFilter.cs

```diff
@@ -265,5 +272,49 @@ private static HttpResponseMessage ConvertResponse(RTHttpResponseMessage rtRespo
 
             return response;
         }
+
+        private static Guid LogRequest(HttpRequestMessage request)
+        {
+            if (_diagnosticSource.IsEnabled("System.Net.Http.Request"))
+            {
+                Guid loggingRequestId = Guid.NewGuid();
+                long timestamp = Stopwatch.GetTimestamp();
+
+                _diagnosticSource.Write(
+                    "System.Net.Http.Request",
+                    new
+                    {
+                        Request = request,
+                        LoggingRequestId = loggingRequestId,
+                        Timestamp = timestamp
+                    });
+
+                return loggingRequestId;
+            }
+
+            return Guid.Empty;
+        }
+
+        private static void LogResponse(HttpResponseMessage response, Guid loggingRequestId)
+        {
+            if (_diagnosticSource.IsEnabled("System.Net.Http.Response"))
+            {
+                // An empty loggingRequestId signifies that the request was not logged, so do
+                // not attempt to log response.
+                if (loggingRequestId != Guid.Empty)
+                {
+                    long timestamp = Stopwatch.GetTimestamp();
+
+                    _diagnosticSource.Write(
+                      "System.Net.Http.Response",
+                      new
+                      {
+                          Response = response,
+                          LoggingRequestId = loggingRequestId,
+                          Timestamp = timestamp
+                      });
+                }
+            }
+        }
```

Comment:
This is because we want to capture direct WinHttpHandler usage, on desktop, as well. If I log at both places then we end up logging the same event twice, and if I do something special to get around the double logging for desktop then we end up with a split between the platforms like I have here already.